builders/interact/ethereum-api/dev-env/foundry.md

```diff
@@ -0,0 +1,359 @@
+
+```bash
+    cast send --private-key INSERT_YOUR_PRIVATE_KEY \
+    --rpc-url https://fraa-dancebox-3001-rpc.a.dancebox.tanssi.network \
+    --chain 5678 \
+    INSERT_YOUR_CONTRACT_ADDRESS \
+    "transfer(address,uint256)" 0x0000000000000000000000000000000000000001 1
+```
+
+The transaction will be signed by your EVM account and be broadcasted to the network. The output should look similar to:
+
+![Foundry Contract Interaction](/images/builders/interact/ethereum-api/dev-environments/foundry/foundry-4.png)
+
+Congratulations, you have successfully deployed and interacted with a contract using Foundry!
+
+## Forking with Anvil {: #forking-with-cast-anvil }
+
+As previously mentioned, [Anvil](https://book.getfoundry.sh/anvil/){target=_blank} is a local TestNet node for development purposes that can fork preexisting networks. Forking the demo EVM ContainerChain allows you to interact with live contracts deployed on the network.
+
+To fork the demo EVM ContainerChain from the command line, you can run the following command from within your Foundry project directory. You can also replace the RPC URL with the RPC URL of your EVM ContainerChain.
+
+```bash
+    anvil --fork-url https://fraa-dancebox-3001-rpc.a.dancebox.tanssi.network
+```
+
+Your forked instance will have 10 development accounts that are pre-funded with 10,000 test tokens. The forked instance is available at `http://127.0.0.1:8545/`. The output in your terminal should resemble the following:
+
+![Forking terminal screen](/images/builders/interact/ethereum-api/dev-environments/foundry/foundry-5.png)
+
+To verify you have forked the network, you can query the latest block number and compare it to the current block number of the [demo EVM ContainerChain](https://tanssi-evmexplorer.netlify.app/){target=_blank}
+
+```bash
+curl --data '{"method":"eth_blockNumber","params":[],"id":1,"jsonrpc":"2.0"}' -H "Content-Type: application/json" -X POST localhost:8545 
+```
+
+If you convert the `result` from [hex to decimal](https://www.rapidtables.com/convert/number/hex-to-decimal.html){target=_blank}, you should get the latest block number from the time you forked the network. 
+
+From here you can deploy new contracts to your forked instance of the demo EVM ContainerChain (or any other EVM ContainerChain) or interact with contracts already deployed. Building off of the previous example in this guide, you can make a call using Cast to check the balance of the minted MYTOK tokens in the account you deployed the contract with:
+
+```bash
+cast call INSERT_CONTRACT_ADDRESS  "balanceOf(address)(uint256)" INSERT_YOUR_ADDRESS --rpc-url http://localhost:8545
+```
+
+## Using Chisel {: #using-chisel }
+
+Chisel is a Solidity REPL, or shell. It allows a developer to write Solidity directly in the console for testing small snippets of code, letting developers skip the project setup and contract deployment steps for what should be a quick process.  
```

Comment:
I like how there was a link for Anvil again at the beginning of the Anvil section. Can you please add a link here (and for other sections if it is missing there)?